synapse/tests/test_cortex.py

```diff
@@ -2849,6 +2849,33 @@ def test_cortex_formtufobytufo(self):
             self.none(t1[1].get('strform:baz'))
             self.none(t1[1].get('strform:haha'))
 
+    def test_cortex_fifos_busref(self):
+        with self.getTestDir() as dirn:
+
+            url = 'dir:///' + dirn
+            with s_cortex.openurl(url) as core:
+                node = core.formTufoByProp('syn:fifo', '(haHA)', desc='test fifo')
+                name = node[1].get('syn:fifo:name')
+
+                # refcount is set to 1 when the fifo is created (when the syn:fifo node is formed),
+                #   then incremented when getCoreFifo is called
+                fiforef_0 = core.getCoreFifo(name)
+                self.eq(2, fiforef_0._syn_refs)
+
+                # Add 3 more refs
+                fiforef_1 = core.getCoreFifo(name)
+                fiforef_2 = core.getCoreFifo(name)
+                fiforef_3 = core.getCoreFifo(name)
+                self.eq(5, fiforef_0._syn_refs)
+
+                # Begin to remove them
+                fiforef_3.fini()
+                self.eq(4, fiforef_0._syn_refs)
+                fiforef_2.fini()
+                self.eq(3, fiforef_0._syn_refs)
+                fiforef_1.fini()
+                self.eq(2, fiforef_0._syn_refs)
+
```

Comment:
We should make sure ``fiforef_0`` is actually fini'd at the end here

Reply:
done